### PR TITLE
StatObject Tests for read only integration tests

### PR DESCRIPTION
**Pull request by Tulsishah**
This will be pull request template
### Description
StatObject Tests
-  statExistingObject - File, Dir , FileFromSubDirectory, SubDirectory
-  statNonExistingObject - File, Dir , FileFromSubDirectory, SubDirectory

### Testing details
1. Manual - Tested all the tests are passing
2. Unit tests
3. Integration tests

### CODE REVIEW

File: tools/integration_tests/readonly/list_objects_test.go [hunk 1]
```diff
@@ -39,11 +39,11 @@ func TestListObjectsInBucket(t *testing.T) {
 		t.Errorf("The number of objects in the current directory doesn't match.")
 	}
 
-	if obj[0].Name() != DirectoryNameInTestBucket && obj[0].IsDir() != true {
+	if obj[0].Name() != DirectoryNameInTestBucket || obj[0].IsDir() != true {
```
Comment (Tulsishah) [inline tools/integration_tests/readonly/list_objects_test.go:42]: For Reviewer: if any of the condition fails, it will return error

Comment (sethiay) [reply tools/integration_tests/readonly/list_objects_test.go:42]: if this is a fix for list tests then please raise a separate PR for it.

File: tools/integration_tests/readonly/stat_object_test.go [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+func TestStatSubDirectory(t *testing.T) {
+	DirPath := path.Join(setup.MntDir(), DirectoryNameInTestBucket, SubDirectoryNameInTestBucket)
+	dir := statExistingObj(DirPath, t)
+
+	if dir.Name() != SubDirectoryNameInTestBucket || dir.IsDir() != true {
+		t.Errorf("Stat incorrrect Directory.")
+	}
+}
+
+func statNotExistingObj(objPath string, t *testing.T) {
```
Comment (sethiay) [inline tools/integration_tests/readonly/stat_object_test.go:70]: Please change the name of this function according to the comment on this PR: https://github.com/GoogleCloudPlatform/gcsfuse/pull/1069/files#r1175119688

File: tools/integration_tests/readonly/list_objects_test.go (hunk 1 above)
Comment (Tulsishah) [reply tools/integration_tests/readonly/list_objects_test.go]: okay

**Pull request closed without merging**